types/index.d.ts

```diff
@@ -0,0 +1,220 @@
+// TypeScript Version: 2.2
+import * as amazon from "./amazon";
+
+export type RequestHandler = (request: request, response: response) => void;
+export type ErrorHandler = (e: any, request: request, response: response) => void;
+export type PreOrPostHandler = (request: request, response: response, type: string) => void;
+
+export let apps: object;
+
+export class app {
+  constructor(name: string);
+
+  pre: PreOrPostHandler;
+  post: PreOrPostHandler;
+
+  name: string;
+  messages: any;
+
+  persistentSession: boolean;
+  exhaustiveUtterances: boolean;
+  error?: ErrorHandler;
+
+  dictionary: any;
+  intents: {[name: string]: amazon.Intent};
+
+  // TODO
+  audioPlayerEventHandlers: any;
+  audioPlayer: (eventName: string, func: any) => void;
+
+  launchFunc?: RequestHandler;
+  launch: (func: RequestHandler) => void;
+
+  sessionEndedFunc?: RequestHandler;
+  sessionEnded: (func: RequestHandler) => void;
+
+  request: (requestJSON: amazon.Request) => void;
+  schema: () => string;
+  utterances: () => string;
+
+  handler: (event: string, context: amazon.Context, callback: (error: Error, response: response) => void) => void;
+  lambda: (event: string, context: amazon.Context, callback: (error: Error, response: response) => void) => void;
+
+  express: (options: ExpressOptions) => void;
+
+  intent: (intentName: string, schema: IntentSchema, handler: RequestHandler) => void;
+}
+
+export class request {
+  type: () => "LaunchRequest"|"IntentRequest"|"SessionEndedRequest";
+
+  // return the value passed in for a given slot name
```

Comment:
Ah, fascinating. When I first poked at doing `/** etc */`, I couldn't get those comments to show up in VS Code for me, so I figured leaving the comments as-is didn't matter (the whole structure is just taken from the API docs in the README). 

Seems like it was a PEBKAC, though — `*doc`-style comments are totally showing up for me now. I'll go through and tidy that up.